File: molecular/src/main/java/com/hartwig/actin/molecular/orange/curation/ExternalTrialMapping.java
package com.hartwig.actin.molecular.orange.curation;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ExternalTrialMapping {

    @NotNull
    public abstract String externalTrial();

    @NotNull
    public abstract String actinTrial();
}

File: molecular/src/main/java/com/hartwig/actin/molecular/orange/evidence/curation/ExternalTrialMapping.kt
package com.hartwig.actin.molecular.orange.evidence.curation

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class ExternalTrialMapping {
    abstract fun externalTrial(): String
    abstract fun actinTrial(): String
}
